### PR TITLE
implement ListOps.filter

### PR DESCRIPTION
**Pull request by odeac**
## Overview
see title

### JIRA ticket:
None

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/resources/ListOps.rho [hunk 1]
```diff
@@ -218,6 +218,30 @@ new  ListOps, rs(`rho:registry:insertSigned:secp256k1`), uriOut in {
     }
   } |
 
+  contract ListOps(@"filter", @list, predicate, returnCh) = {
+    new reverseCh, accumulate, stdout(`rho:io:stdout`) in {
```
Comment (JosephDenman) [inline casper/src/main/resources/ListOps.rho]: Stdout

**Review by JosephDenman — CHANGES_REQUESTED** [review]
You have debugging leftovers. Otherwise, LGTM.

File: casper/src/main/resources/ListOps.rho (hunk 1 above)
Comment (odeac) [reply casper/src/main/resources/ListOps.rho]: the horror!!!!!

**Review by JosephDenman — APPROVED** [review]

**Review by KentShikama — APPROVED** [review]

**Pull request merged**